hardware/CommunicationsPi/web_client.py

```diff
@@ -27,15 +27,20 @@ def get_server_url_from_env(self):
         return url
 
     # Function to ping the LAN server
-    # Accepts payload as a python dictionary
-    def ping_lan_server(self, payload):
-        self.logging.info("Pinging")
+    # Accepts payload as a python dictionary or json object
+    def send(self, payload, is_json=True):
+        self.logging.info("Pinging: " + self.url)
 
         try:
-            self.logging.info("data: " + payload)
-            response = requests.post(self.url, data=payload)
-            response.raise_for_status()
-            return response
+            self.logging.info("data: " + str(payload))
+            if is_json:
+                response = requests.post(self.url, json=payload)
+                response.raise_for_status()
+                return response
+            else:
+                response = requests.post(self.url, data=payload)
```

Comment:
apparently with requests.post, using the json parameter automatically sets the Content-Type to application/json. Is the remote app expecting a certain content type? If we are setting the Content-Type on line 41 implicitly should we also be setting it explicitly in 37-38?